src/matrix.rs

```diff
@@ -9,9 +9,26 @@ macro_rules! define_square_matrix_struct {
         }
 
         impl $name {
-            fn new(data: [[f64; $order]; $order]) -> Self {
+            pub fn new(data: [[f64; $order]; $order]) -> Self {
                 Self { data }
             }
+
+            pub fn zero() -> Self {
+                let data: [[f64; $order]; $order] = [[0.0; $order]; $order];
+                Self::new(data)
+            }
+
+            pub fn identity() -> Self {
+                let mut result = Self::zero();
+                for r in 0..$order {
+                    for c in 0..$order {
+                        if r == c {
+                            result[r][c] = 1.0;
+                        }
+                    }
+                }
```

Comment:
Could do it with only one loop:
```suggestion
                for i in 0..$order {
                   result[i][i] = 1.0;
                }
```